js-core/homeworks/homework-4/src/main.js

```diff
@@ -0,0 +1,130 @@
+/*
+* Напишите функцию которая будет принимать 1 аргумент и в зависимости от типа аргумента
+* Если тип аргумента число или объект -> возвращать true
+* Если тип аргумента функция -> возвращать false
+* Если тип аргумента Строка и длина этой строки не равна 10 -> возвращать "длина вашей строки: <длина строки>
+* Если длина 10 -> 'you win'
+* */
+let objCheck = {
+    name: 'aloe'
+};
+let numbCkeck = 15;
+let stringCheck = 'aloe vera lorem ipsum';
+let stringTenCheck = 'aloe vera.';
+let funcCheck = function(){
+    return 'yes it is';
+};
+
+function check(param){
+    let numOrObject = typeof param === 'number' || typeof param === 'object';
+    let functionType = typeof param === 'function';
+    let stringType = typeof param === 'string' && param.length != 10;
+    let stingTenType = typeof param === 'string' && param.length === 10;
+
+    if(numOrObject) return true;
+    else if(functionType) return false;
+    else if(stringType) return 'Длина вашей строки:' + param.length;
+    else if(stingTenType) return 'you win';
+
+    return param;
+}
+
+console.log(check(numbCkeck));
+console.log(check(objCheck));
+console.log(check(funcCheck));
+console.log(check(funcCheck));
+console.log(check(stringCheck));
+console.log(check(stringTenCheck));
+
+//==============================================================================================
+
+/*
+* Напишите функцию которая принимает 2 числа
+* и возвращает массив содержащий числа между первым числом и вторым числом;
+ */
+
+
+ function betweenNumbers(num1,num2){
+    let arr = [];
+    for(let i = num1; i < num2 + 1; i++){
+    arr[i] = i;
+    };
+    arr.splice(0,arr.indexOf(num1));
```

Comment:
can we simplify it?

You have to solve it without any array method. There much easier approach